### ISSUE TITLE
random thoughts

### ISSUE THREAD

**Issue by cgwalters** [runcom/kiosk-sagano-example#1]
One thing I think we probably want here is for the kiosk user to be transient, right?

I know GDM had "guest user" support long ago that was basically this...

Nowadays of course there's also systemd `DynamicUser=` that we could potentially attach to `gdm.service` or so?  It's meant for system, not login users but it might be enough.

I think it'd be good to try to set `HOME=/usr/share/kiosk-user` instead of having any persistent+writable state in `/var` at all.  Or if we need writability, a transient overlayfs would likely suffice, right?

**Comment by runcom** [comment]
yes to all the above, I've assembled this to explore shortcomings mainly as I would have loved to just `useradd` and similar things (also setting the default target was kind of expected to me)

**Comment by runcom** [comment]
> Nowadays of course there's also systemd `DynamicUser=` that we could potentially attach to `gdm.service` or so? It's meant for system, not login users but it might be enough.
> 
> I think it'd be good to try to set `HOME=/usr/share/kiosk-user` instead of having any persistent+writable state in `/var` at all. Or if we need writability, a transient overlayfs would likely suffice, right?

I could explore these too in the future although I fear they're gonna mess with automatic login and such but I'll try (I've already run into some issues)
